src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -172,21 +172,38 @@ internal static void ValidateNumber(ReadOnlySpan<byte> utf8FormattedNumber)
                 return;
             }
 
+            //The non digit character inside the number
             byte val = utf8FormattedNumber[i];
 
             if (val == '.')
             {
                 i++;
+
+                while (i < utf8FormattedNumber.Length && JsonHelpers.IsDigit(utf8FormattedNumber[i]))
+                {
+                    i++;
+                }
             }
-            else if (val == 'e' || val == 'E')
+
+            if (utf8FormattedNumber.Length <= i)
+            {
+                throw new ArgumentException(SR.RequiredDigitNotFoundEndOfData, nameof(utf8FormattedNumber));
```

Comment:
Should these be ThrowHelper like elsewhere in the file?

Reply:
ThrowHelper is usually used to change the inlining cost/value of a method.  If there's not already a ThrowHelper for this, it's probably not worth adding one.  I don't think there is, because this is "reader logic", so it'd be in the reader's ThrowHelper, which requires passing in a ref reader to explain how you go to the bad state.

In general, we prefer `throw new` over ThrowHelper, doing the latter only where needed/justified.

Everything else in this file is a very small method that is trying to be an inline candidate.  This is probably too big to inline even with ThrowHelper.